asistente_ladm_col/gui/controlled_measurement_dialog.py

```diff
@@ -114,31 +129,151 @@ def accept_dialog(self):
                 x_list.append(current_point.x())
                 y_list.append(current_point.y())
 
-            x_mean = statistics.mean(x_list)
-            y_mean = statistics.mean(y_list)
-            x_stdev = statistics.pstdev(x_list)
-            y_stdev = statistics.pstdev(y_list)
+            if x_list and y_list != []:
+                x_mean = statistics.mean(x_list)
+                y_mean = statistics.mean(y_list)
+                x_stdev = statistics.pstdev(x_list)
+                y_stdev = statistics.pstdev(y_list)
+            else:
+                continue
             geom = QgsGeometry.fromPointXY(QgsPointXY(x_mean, y_mean))
-
-            new_feature = QgsVectorLayerUtils.createFeature(layer, geom,
-                {
-                    0: group_id,
-                    1: len(x_list),
-                    2: x_mean,
-                    3: y_mean,
-                    4: x_stdev,
-                    5: y_stdev
-                }
-            )
+            fields_values.update({
+                layer.fields().indexOf("group_id"): group_id,
+                layer.fields().indexOf("count"): len(x_list),
+                layer.fields().indexOf("x_mean"): x_mean,
+                layer.fields().indexOf("y_mean"): y_mean,
+                layer.fields().indexOf("x_stdev"): x_stdev,
+                layer.fields().indexOf("y_stdev"): y_stdev
+            })
+            new_feature = QgsVectorLayerUtils.createFeature(layer, geom, fields_values)
             new_features.append(new_feature)
 
         layer.dataProvider().addFeatures(new_features)
+        features = groups.getFeatures("\"{}\" IS NULL".format("belongs_to_group"))
+        layer.dataProvider().addFeatures(features)
+        layer.commitChanges()
         QgsProject.instance().addMapLayer(layer)
 
         self.qgis_utils.message_emitted.emit(
             QCoreApplication.translate("ControlledMeasurementDialog",
                                        "A new average point layer has been added to the map!"),
             Qgis.Info)
 
+    def run_group_points_model(self, input_layer, tolerance, definition_field):
+        # Run model
+        model = QgsApplication.processingRegistry().algorithmById("model:Group_Points")
+        if model:
+            params = {
+                '1_Inputpoints': input_layer.source(),
+                '2_Typedefinition': definition_field,
+                '3_Tolerance': tolerance,
+                'native:multiparttosingleparts_2:output': 'memory:',
+                'native:mergevectorlayers_1:output': 'memory:'
+            }
+            res = processing.run("model:Group_Points", params)
+            msg = "Model Group_Points and execute OK!"
+            return res, msg
+        else:
+            res = None
+            msg = "Model Group_Points was not found and cannot be opened!"
+            return res, msg
+
+    def time_validate(self, input_features, time_tolerance, time_field):
```

Comment:
This function does a lot of things, could you please add some documentation/description of what it does (perhaps per code blocks)?

Reply:
solved in the commit: https://github.com/AgenciaImplementacion/Asistente-LADM_COL/commit/41a6511b624039fa0bb91702dbef2206a0e1bc83